rclcpp/include/rclcpp/subscription.hpp

```diff
@@ -107,6 +108,14 @@ class Subscription : public SubscriptionBase
       this->add_event_handler(event_callbacks.liveliness_callback,
         RCL_SUBSCRIPTION_LIVELINESS_CHANGED);
     }
+    TRACEPOINT(
+      rclcpp_subscription_callback_added,
+      (const void *)get_subscription_handle().get(),
+      (const void *)&any_callback_);
+    // TODO(christophebedard) find another way
+    // this object seems to get copied, so if registration is done too early
+    // (e.g. in `set()`), `this` won't match any address used later
+    any_callback_.register_callback_function();
```

Comment:
@iluetkeb I don't see other reasonable way either, to cope with `std::function` copies I mean. In which case I'm reluctant to keep that TODO, as opposed to clearly documenting that tracepoints should not be registered till `AnyCallback` has hit "stable" storage.

Reply:
@christophebedard any objections to removing that?

Reply:
sure, no problem!

Reply:
Yes, please. Remove that TODO **AND** document the caveats of the tracing system w.r.t. memory copies.